cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/Index.java

```diff
@@ -61,7 +61,7 @@ private Index(List<Object> fieldNames,
      * This method sets the index type to the default setting of "json"
      *
      * @param fieldNames the field names in the index
-     * @param indexName the index name
+     * @param indexName the index name or null
```

Comment:
I don't think its right to to say what happens in the `null` case when creating an index, as saying it here would make the user think that the indexname is generated when the Index object is created, which it isn't, so it would be misleading to the user if they were trying to create their own `Index` object.